experimental/dds/tree2/src/feature-libraries/editable-tree-2/editableTreeContext.ts

```diff
@@ -0,0 +1,158 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { assert } from "@fluidframework/common-utils";
+import {
+	IEditableForest,
+	moveToDetachedField,
+	ForestEvents,
+	FieldStoredSchema,
+	FieldKey,
+} from "../../core";
+import { ISubscribable } from "../../events";
+import { DefaultEditBuilder } from "../default-field-kinds";
+import { NodeKeyManager } from "../node-key";
+import { FieldGenerator } from "../contextuallyTyped";
+import { TypedSchemaCollection } from "../typed-schema";
+import { UntypedField } from "./editableTreeTypes";
+import { makeField } from "./lazyField";
+import { LazyEntity } from "./lazyEntity";
+
+/**
+ * A common context of a "forest" of EditableTrees.
+ * It handles group operations like transforming cursors into anchors for edits.
+ * @alpha
+ */
+export interface TreeContext extends ISubscribable<ForestEvents> {
+	/**
+	 * Gets the root field of the tree.
+	 */
+	get root(): UntypedField;
+
+	/**
+	 * Schema used within this context.
+	 * All data must conform to these schema.
+	 */
+	readonly schema: TypedSchemaCollection;
+
+	/**
```

Comment:
Nit: My recommendation for structuring comments with this kind of structure:

```typescript
	/**
	 * Call before editing.
	 *
	 * @remarks Note that after performing edits, EditableTrees for nodes that no longer exist are invalid to use.
	 * 
	 * @privateRemarks TODO: maybe add an API to check if a specific EditableTree still exists,
	 * and only make use other than that invalid.
	 */
```

Reply:
Also nit: Can we also elaborate more on what this does / why it needs to be called before editing?

Reply:
This comment (and these details) are unchanged from the old editable tree. Probably don't look at this file yet.

Reply:
anyway, the reason here is that cursors are illegal to hold onto across edits, and this cleans them up.